### PR TITLE
Eval blacklist

### PR DESCRIPTION
**Pull request by jvoisin**
This is an implementation of whitelist/blacklist for `eval`, currently work in progress.
 
- [x] Allow to blacklist a single function in `eval`
- [x] Allow a list of functions to be blacklisted
- [x] Add tests for nested `eval`
- [x] Add simulation mode
- [x] Add documentation

### CODE REVIEW

File: src/sp_config.h [hunk 1]
```diff
@@ -111,6 +111,12 @@ typedef struct {
   sp_cidr *cidr;
 } sp_disabled_function;
 
+typedef struct {
+  sp_list_node *blacklist;
+  sp_list_node *whitelist;
+	bool simulation;
```
Comment (blotus) [inline src/sp_config.h]: Bad indentation

File: src/sp_disabled_functions.c [hunk 2]
```diff
@@ -460,6 +460,28 @@ static int hook_functions(const sp_list_node* config) {
...
 }
 
+ZEND_FUNCTION(eval_filter_callback) {
+  void (*orig_handler)(INTERNAL_FUNCTION_PARAMETERS);
+  const char* current_function_name = get_active_function_name(TSRMLS_C);
+
+  if (SNUFFLEUPAGUS_G(in_eval) == true) {
+		if (1 == SNUFFLEUPAGUS_G(config).config_eval->simulation) {
+			sp_log_msg("eval", SP_LOG_SIMULATION,
+					"A call to %s was tried in eval, droping it.", current_function_name);
```
Comment (blotus) [inline src/sp_disabled_functions.c]: Dropping, not droping

File: src/sp_disabled_functions.c [hunk 3]
```diff
@@ -460,6 +460,28 @@ static int hook_functions(const sp_list_node* config) {
...
+		} else {
+			sp_log_msg("eval", SP_LOG_DROP,
+					"A call to %s was tried in eval, droping it.", current_function_name);
```
Comment (blotus) [inline src/sp_disabled_functions.c]: Same.

File: src/sp_disabled_functions.c [hunk 4]
```diff
@@ -460,6 +460,29 @@ static int hook_functions(const sp_list_node* config) {
...
 }
 
+ZEND_FUNCTION(eval_filter_callback) {
+  void (*orig_handler)(INTERNAL_FUNCTION_PARAMETERS);
+  const char* current_function_name = get_active_function_name(TSRMLS_C);
+
+  if (SNUFFLEUPAGUS_G(in_eval) == true) {
+    if (1 == SNUFFLEUPAGUS_G(config).config_eval->simulation) {
+      sp_log_msg("eval", SP_LOG_SIMULATION,
+                 "A call to %s was tried in eval, dropping it.",
```
Comment (buixor) [inline src/sp_disabled_functions.c]: Can we log the file / line where it was called ?

**Pull request merged**